keps/sig-api-machinery/2019-02-13-custom-resource-definitions.md

```diff
@@ -0,0 +1,212 @@
+---
+kep-number: 0
+title: Custom Resource Definitions
+authors:
+  - "@brendandburns"
+owning-sig: sig-api-machinery
+participating-sigs:
+  - sig-architecture
+reviewers:
+  - "@bgrant0607"
+  - "@smarterclayton"
+  - "@lavalamp"
+  - "@deads2k"
+approvers:
+  - "@lavalamp"
+  - "@smarterclayton"
+  - TBD
+editor: TBD
+creation-date: "2013-02-13"
+last-updated: "2013-02-13"
+status: implemented
+---
+
+# Custom Resource Definitions
+
+## Summary
+
+This KEP covers the addition and usage of Custom Resource Definitions. This KEP is different than most KEPs
+because it is an attempt to retrospectively describe the development of a feature in Kubernetes that 
+already exists. We are doing this because custom resource definitions were created and evolved before the
+KEP process was put in place. Because of this, there is no single place that describes the reasons for
+custom resource definitions and how they should be used in Kubernetes.
+
+## Motivation
+
+The primary motivation for custom resources was to provide a simple easy to use extension mechanism
+for Kubernetes clusters to enable customization by Kubernetes users and to facilitate a rich
+ecosystem of plugins for Kubernetes clusters. If Kubernetes is the kernel, CustomResourceDefinitions
+are intended to be loadable kernel modules.
+
+### Goals
+
+   * Provide an easy to use mechanism for the dynamic addition of new API types to Kubernetes.
```

Comment:
What I miss: It should not be observable whether a resource is a CustomResources or natively implemented (other than looking at the CRD objects). In other words: same semantics as native resources.